offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/offergeneration/QuotationDetails.groovy

```diff
@@ -377,14 +294,23 @@ class QuotationDetails {
          * @return returns the HTML code as string
          */
         static String itemInHTML(int offerPosition, ProductItem item) {
-            String totalCost = Currency.getFormatterWithoutSymbol().format(item.quantity * item.product.unitPrice)
+            String totalCost = Currency.getFormatterWithoutSymbol().format(item.getTotalCost())
+            double unitPrice = item.product.externalUnitPrice
+            if(offer.isInternal()) {
+              unitPrice = item.product.internalUnitPrice
+            }
+            String discountRow = ""
+            if(item.product.discount > 0) {
+              discountRow = """<div class="col-2 price-value">${-Currency.getFormatterWithoutSymbol().format(item.product.discount)}</div>"""
+            }
             return """<div class="row product-item">
                         <div class="col-1">${offerPosition}</div>
                         <div class="col-4 ">${item.product.productName}</div>
                         <div class="col-1 price-value">${item.quantity}</div>
                         <div class="col-2 text-center">${item.product.unit}</div>
-                        <div class="col-2 price-value">${Currency.getFormatterWithoutSymbol().format(item.product.unitPrice)}</div>
-                        <div class="col-2 price-value">${totalCost}</div>
+                        <div class="col-2 price-value">${Currency.getFormatterWithoutSymbol().format(unitPrice)}</div>"""
+                        + discountRow +
+                        """<div class="col-2 price-value">${totalCost}</div>
```

Comment:
why adding the discount row here? 
There are several things you should consider here:
1. You are adding the row between the columns of the unit price and the total cost (and I assume you want it to be a row, right?)
2. If you want to add is as a column you need to adapt the current table, col-2 tells the table how large the column is, one row has a space of 12, you need to reorganize the column content if you want to do that (assign new spaces for the column to make room for the new column)
3. I would not add the discount row as you did it but rather stick to the style used before and add the row in groovy style ${YOUR_VARIABLE}. I would leave all the HTML stuff here and just transfere the value for the discount

Reply:
1. I think this is where it would fit best? Then you have "base price per unit", "how much discount there was", "the resulting price for that product", right?
2. I am quite confused about these being columns, you are of course right that I wanted to add these as a row. I am open for suggestions!